arangosh/CMakeLists.txt

```diff
@@ -43,15 +43,19 @@ install(
 install_config(arangobench)
 
 if (NOT USE_PRECOMPILED_V8)
-  add_dependencies(arangobench zlibstatic v8_build) # v8_build includes ICU build
+  add_dependencies(${BIN_ARANGOBENCH} zlibstatic v8_build) # v8_build includes ICU build
```

Comment:
Please remove the extra indirections (the others below as well).

from ProfessionalCMake_1.0.3.pdf which the company bought (I think there is some other place where it mentions target names explicitly but that was close enough for me)

## 4.6. Recommended Practices
Target names need not be related to the project name. It is common to see tutorials and example use  a  variable  for  the  project  name  and  reuse  that  variable  for  the  name  of  an  executable  target like so:

```CMake
# Poor practice, but very common
set(projectName MyExample)
project(${projectName})
add_executable(${projectName}...)
```

This only works for the most basic of projects and encourages a number of bad habits. Consider the project  name  and  executable  name  as  being  separate,  even  if  initially  they  start  out  the  same.  **Set the  project  name  directly  rather  than  via  a  variable,**  choose  a  target  name  according  to  what  the target does rather than the project it is part of and assume the project will eventually need to define more than one target. This reinforces better habits which will be important when working on more complex multi-target projects